source/fuzz/fuzzer_pass_permute_function_variables.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_FUZZER_PASS_PERMUTE_FUNCTION_VARIABLES_H_
+#define SOURCE_FUZZ_FUZZER_PASS_PERMUTE_FUNCTION_VARIABLES_H_
+
+#include "source/fuzz/fuzzer_pass.h"
+#include "source/fuzz/transformation_swap_function_variables.h"
+
+namespace spvtools {
+namespace fuzz {
+
+// A pass that permute function variables.
+// this is final class so, can't be inherited. 
+class FuzzerPassPermuteFunctionVariables  final: public FuzzerPass {
```

Comment:
We have not made other subclasses of FuzzerPass final. There could be an argument for doing so, but given that we haven't done it so far I would prefer for this class to be non-final. (If we decide we want them to be final we should do that all at once in a single PR.)

Reply:
I've removed it.